File: test/Class.test.ts
import {expect} from 'chai';
import {Point} from "../src/class/Point";

describe('Class test', () => {
  
  it('It should create a point', () => {
    let point: Point = new Point(0,0);
    
    expect(point.constructor.name).to.equal(Point.name);
  });
  
  it('It should create a 2d point', () => {
    let point: Point = new Point(3,7);
    
    let sum: number = point.x + point.y;
    
    expect(sum).to.equal(10);
  });
  
  it('It should create a 3d point', () => {
    let point: Point = new Point(10, 5, 5);
    
    let sum: number = point.x + point.y + point.z;
    
    expect(sum).to.equal(20);
  });
  
});

File: test/src/ClassTest.kt
import aclass.Point
import org.testng.AssertJUnit.assertEquals
import org.testng.annotations.Test

class ClassTest{

  @Test
  fun `It should create a point`(){
    val point: Point = Point(0f, 0f)

    assertEquals(point.javaClass.simpleName, Point::class.simpleName)
  }

  @Test
  fun `It should create a 2d point`(){
    val point: Point = Point(3f, 7f)
    val sum: Float = point.x + point.y

    assertEquals(sum, 10f, 0.01f)
  }

  @Test
  fun `It should create a 3d point`(){
    val point: Point = Point(10f, 5f, 5f)
    val sum: Float = point.x + point.y + point.z

    assertEquals(sum, 20f, 0.01f)
  }
}